lib/puppet/pops/loader/module_loaders.rb

```diff
@@ -214,8 +214,12 @@ def initialize(parent_loader, loaders, module_name, path, loader_name, loadables
     end
 
     def existing_path(effective_path)
+      expanded_path = Puppet::FileSystem.expanded_path_string(effective_path)
```

Comment:
Here I am thinking, that it would be better if `effective_path` was already an "implementation specific path" since that is the purpose of this entire mechanism (comparing a tentative path against an existing path obtained from Dir.glob). Now the logic would first compose a non file system specific path out of "a root for a module or env" + a number of directory "segments" (for example 'functions', or 'type'). The construction of those segments into the "effective path" could perhaps do the "impl specific path" right away using fewer intermediate objects and scans over the characters in the same (or very similar) strings.